kernel/module.c

```diff
@@ -1827,6 +1828,7 @@ static void unset_module_init_ro_nx(struct module *mod) { }
 void __weak module_free(struct module *mod, void *module_region)
 {
 	vfree(module_region);
+	kasan_module_free(module_region);
```

Comment:
This is a wrong place to insert the call. module_free is __weak and intended as default implementation unless arch wants to override it. A bunch of arch in fact override it.